GANDLF/models/seg_modules/DecodingModule.py

```diff
@@ -8,85 +8,54 @@ def __init__(
         self,
         input_channels,
         output_channels,
-        Conv,
-        Dropout,
-        InstanceNorm,
-        leakiness=1e-2,
-        conv_bias=True,
-        kernel_size=3,
-        inst_norm_affine=True,
-        res=True,
-        lrelu_inplace=True,
+        conv=nn.Conv2d,
+        conv_kwargs=None,
+        norm=nn.BatchNorm2d,
+        norm_kwargs=None,
+        act=nn.LeakyReLU,
+        act_kwargs=None,
+        network_kwargs=None,
     ):
-        """[The Decoding convolution module to learn the information and use later]
+        nn.Module.__init__(self)
+        if conv_kwargs is None:
+            conv_kwargs = {"kernel_size": 3, "stride": 1, "padding": 1, "bias": True}
+        if norm_kwargs is None:
+            norm_kwargs = {
+                "eps": 1e-5,
+                "affine": True,
+                "momentum": 0.1,
+                "track_running_stats": True,
+            }
+        if act_kwargs is None:
+            act_kwargs = {"negative_slope": 1e-2, "inplace": True}
+        if network_kwargs is None:
+            network_kwargs = {"res": False}
 
-        [This function will create the Learning convolutions]
+        self.res = network_kwargs["res"]
 
-        Arguments:
-            input_channels {[int]} -- [the input number of channels, in our case
-                                       the number of channels from downsample]
-            output_channels {[int]} -- [the output number of channels, will det-
-                                        -ermine the upcoming channels]
+        self.conv0 = conv(input_channels, output_channels, **conv_kwargs)
+        self.conv1 = conv(output_channels, output_channels, **conv_kwargs)
+        self.conv2 = conv(output_channels, output_channels, **conv_kwargs)
 
-        Keyword Arguments:
-            kernel_size {number} -- [size of filter] (default: {3})
-            leakiness {number} -- [the negative leakiness] (default: {1e-2})
-            conv_bias {bool} -- [to use the bias in filters] (default: {True})
-            inst_norm_affine {bool} -- [affine use in norm] (default: {True})
-            res {bool} -- [to use residual connections] (default: {False})
-            lrelu_inplace {bool} -- [To update conv outputs with lrelu outputs]
-                                    (default: {True})
```

Comment:
Since this is making extensive use of kwards, we need to keep detailed documentation.

Reply:
That is where you come in my friend. 🤣

Reply:
Hmm... 😆 

I would imagine it would make more sense for you to write the documentation, since you are making the code changes...